istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,510 @@
+# Display the webhook configuration of Galley
+istioctl experimental post-install webhook status --validation --config-name istio-galley
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !opts.validationWebhook {
+				fmt.Println("not displaying validation webhook")
+				return nil
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				fmt.Printf("err when creating k8s client interface: %v\n", err)
+				return err
+			}
+			config, err := getValidatingWebhookConfig(client, opts.validatingWebhookConfigName)
+			if err != nil {
+				fmt.Printf("error getting validatingwebhookconfiguration: %v\n", err)
+				return err
+			}
+			b, err := yaml.Marshal(config)
+			if err != nil {
+				fmt.Printf("error when marshaling validatingwebhookconfiguration to yaml: %v\n", err)
+				return err
+			}
+			fmt.Printf("validatingwebhookconfiguration %v is:\n%v\n", opts.validatingWebhookConfigName, string(b))
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.validationWebhook, "validation", true, "Specifies whether displaying"+
+		"the validating webhook configuration.")
+	flags.StringVar(&opts.validatingWebhookConfigName, "config-name", "istio-galley",
+		"The name of the ValidatingWebhookConfiguration to display.")
+
+	return cmd
+}
+
+// Create the validatingwebhookconfiguration
+func createValidatingWebhookConfig(k8sClient kubernetes.Interface,
+	config *v1beta1.ValidatingWebhookConfiguration) (*v1beta1.ValidatingWebhookConfiguration, error) {
+	var whConfig *v1beta1.ValidatingWebhookConfiguration
+	var err error
+	if config == nil {
+		return nil, fmt.Errorf("validatingwebhookconfiguration is nil")
+	}
+	client := k8sClient.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations()
+	_, err = client.Get(config.Name, metav1.GetOptions{})
+	if err == nil {
+		fmt.Printf("update webhook configuration %v\n", config.Name)
+		whConfig, err = client.Update(config)
+	} else {
+		fmt.Printf("create webhook configuration %v\n", config.Name)
+		whConfig, err = client.Create(config)
+	}
+	return whConfig, err
+}
+
+// Delete the validatingwebhookconfiguration
+func deleteValidatingWebhookConfig(k8sClient kubernetes.Interface, name string) error {
+	client := k8sClient.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations()
+	err := client.Delete(name, &metav1.DeleteOptions{})
+	return err
+}
+
+// Get the validatingwebhookconfiguration
+func getValidatingWebhookConfig(k8sClient kubernetes.Interface, name string) (*v1beta1.ValidatingWebhookConfiguration, error) {
+	client := k8sClient.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations()
+	return client.Get(name, metav1.GetOptions{})
+}
+
+// Read CA certificate and check whether it is a valid certificate.
+func readCACert(client kubernetes.Interface, certPath, secretName, secretNamespace string, maxWaitTime time.Duration) ([]byte, error) {
+	var caCert []byte
+	var err error
+	readLocal := len(certPath) > 0
+	if readLocal {
```

Comment:
The extra `readLocal` doesn't add much since its used immediately afterwards once. You can add the check directly in the conditional, i.e.

```golang
if certPath == "" {
   // local file
} else {
   // check SA
}

Reply:
Done.